src/manifolds/HyperbolicHyperboloid.jl

```diff
@@ -423,9 +424,10 @@ function Random.rand!(
     σ=one(eltype(pX)),
 ) where {N}
     if vector_at === nothing
-        a = σ .* randn(rng, N)
-        pX[1:(end - 1)] .= a
-        pX[end] = sqrt(1 + dot(a, a))
+        a = randn(rng, N)
+        f = 1 + σ * abs(randn(rng))
+        pX[1:(end - 1)] .= a .* sqrt(f^2 - 1) / norm(a)
```

Comment:
Should this use `firstindex`?
```suggestion
        pX[firstindex(pX):(end - 1)] .= a .* sqrt(f^2 - 1) / norm(a)
```

Reply:
Sure, we can use `firstindex`, though we currently don't really care much about supporting non-1-based indexing in Manifolds.jl.